overlord/ifacestate/udevmonitor/udevmon.go

```diff
@@ -50,13 +49,21 @@ type Monitor struct {
 	monitorStop   chan struct{}
 	netlinkErrors chan error
 	netlinkEvents chan netlink.UEvent
+
+	// devices lookup keeps track of all observed devices to know
+	// when to ignore a spurious event (in case it happens, e.g. when device gets reported by both
+	// the enumeration and monitor on startup).
+	// the lookup is based on device paths which are guaranteed to be unique and stable till device gets removed.
+	// the lookup is not persisted and gets populated and updated in response to enumeration and hotplug events.
```

Comment:
It is unclear how a key in a map may be updated but perhaps I'm reading it wrong.

Reply:
We don't update it, only insert or remove keys as devices get added/removed.